File: Meraki_practice/practice.rb
#!/usr/bin/ruby
###############################################################################
# Name        : practice.rb                                                   #
# Author      : Abel Gancsos                                                  #
# Version     : v. 1.0.0.0                                                    #
# Description : Practice working with data structures and sorting.            #
###############################################################################

class Practice
	@list = nil;@mapping=nil;
	def initialize(params={})
		@list = params["--list"] != nil ? params["--list"] : [1, 2, 4, 6, 7, 8, 1, 4, 2];
		@mapping = {};
		@list.sort;
		for i in 0..@list.length - 1
			@mapping[@list[i]] = @mapping[@list[i]] == nil ? 1 : @mapping[@list[i]] + 1;
		end
	end

	def convert_to_binary(value)
		result = "";
		for part in value.split(".")
			result += "." if result != "";
			result += part.to_i.to_s(2).rjust(8, "0");
		end
		return result;
	end

	def get_lowest_repeated()
		@mapping.sort_by {|k, v| k};
		return @mapping.keys[0];
	end

	def get_highest_repeated()
		@mapping.sort_by {|k, v| k}
		return @mapping.keys[@mapping.keys.length - 1];
	end

	def get_least_repeated()
		@mapping.sort_by {|k, v| v}
		return @mapping.keys[0];
	end

	def get_most_repeated()
		@mapping.sort_by {|k, v| v}
		return @mapping.keys[@mapping.keys.length - 1];
	end

	def invoke()
		ip = "127.0.0.1";
		print "IP: #{ip}; #{self.convert_to_binary(ip)}\n";
		print "#{@mapping}\n";
		print "Lowest repeated value: #{self.get_lowest_repeated} = #{@mapping[self.get_lowest_repeated]}\n";
		print "Highest repeated value: #{self.get_highest_repeated} = #{@mapping[self.get_highest_repeated]}\n";
		print "Lest repeated value: #{self.get_least_repeated} = #{@mapping[self.get_least_repeated]}\n";
		print "Most repeated value: #{self.get_most_repeated} = #{@mapping[self.get_most_repeated]}\n";
	end 	
end

params = {};
for i in 0..ARGV.length - 1
	params[ARGV[i]] = ARGV[i + 1];
end

session = Practice.new(params);
session.invoke();

File: Meraki_practice/practice.py
#!/usr/bin/env python3
###############################################################################
# Name        : practice.py                                                   #
# Author      : Abel Gancsos                                                  #
# Version     : v. 1.0.0.0                                                    #
# Description : Practice working with data structures and sorting.            #
###############################################################################
import os, sys;

class Practice:
	ls = None; mapping = None;
	def __init__(self, params=dict()):
		self.ls = params["--list"] if "--list" in params.keys() else [1, 2, 4, 6, 7, 8, 1, 4, 2];
		self.mapping = {};
		for key in self.ls: self.mapping[key] = self.mapping[key] + 1 if key in self.mapping.keys() else 1;
		pass;
	def convert_to_binary(self, value):
		result = "";
		for part in value.split("."):
			if (result != ""): result += ".";
			result += bin(int(part)).replace("b", "").rjust(8, "0");
		return result;
		pass;
	def get_lowest_repeated(self):
		sorted_map = sorted(self.mapping.keys(), key=lambda x:x);
		return sorted_map[0];
	def get_highest_repeated(self):
		sorted_map = sorted(self.mapping.keys(), key=lambda x:x);
		return sorted_map[len(sorted_map) - 1];
	def get_least_repeated(self):
		sorted_map = sorted(self.mapping);
		return sorted_map[0];
	def get_most_repeated(self):
		sorted_map = sorted(self.mapping);
		return sorted_map[len(sorted_map) - 1];
	def invoke(self):
		ip = "127.0.0.1";
		print("IP: {0}; {1}".format(ip, self.convert_to_binary(ip)));
		print("{0}".format(self.mapping));
		print("Lowest repeated value: {0} = {1}".format(self.get_lowest_repeated(), self.mapping[self.get_lowest_repeated()]));
		print("Highest repeated value: {0} = {1}".format(self.get_highest_repeated(), self.mapping[self.get_highest_repeated()]));
		print("Least repeated value: {0} = {1}".format(self.get_least_repeated(), self.mapping[self.get_least_repeated()]));
		print("Most repeated value: {0} = {1}".format(self.get_most_repeated(), self.mapping[self.get_most_repeated()]));
		pass;
	pass;

if __name__ == "__main__":
	params = dict();
	for i in range(0, len(sys.argv) - 1, 2): params[sys.argv[i]] = sys.argv[i + 1];
	session = Practice(params);
	session.invoke();
	pass;
